### DISCUSSION TITLE
Install Instructions for G4 PowerBooks/iMacs/PowerMacs/mac Minis?

### DISCUSSION THREAD

**Discussion by iamhumanipromise** [kth5/archpower#48] [Q&A] [answered]
Hello! First off, thank you! This is so great! Arch is amazing and I am grateful to be able to have it for my old PowerPC hardware, and my future TALOS Raptor system!!!!! 

With expressing my utmost joy out of the way, I shall proceed with my question. 

_Dost thou dear fellow PowerPC Users have a modified Arch Wiki guide to install once USB booted?_ 

**So far I have (For a PowerBook G4 Aluminum 12" 1GHz):** 

- Use dd to write iso to USB (For GUI use Rufus on Windows, Balena Etcher macOS, Linux, Windows) 
- Hold "Option" on PowerBook to boot to OpenFirmware boot device selection menu
- Set Date
- ethernet connection must be set to "up" 
- add 1.1.1.1 or 1.0.0.1 to /etc/resolve.conf (or your own preferred DNS) 
- systemctl restart systemd-networkd 
- pacman -Syy should now work for latest packages 

-------

After this I am lost. Hoping for some help from folks who use this. Perhaps there is a Discord or Telegram group or Reddit post but I reckon going straight to the source here with git could yield results! 

- **What partitions and partition types are needed to boot?**
- **What additional Apple specific configuration is needed?**
- **Anything special need to happen with mkinitcpio initially, or after install to ensure it is correctly updated everytime the kernel is updated via pacman?**

I notice there was a PowerPC specific option, but unsure if needed for this and if so, how to format, how to config, etc? 

-----

Once disk config and grub config are out of the way, everything else *should* theoretically be smooth sailing!

**Comment by kth5** [comment, marked as answer]
I have this issue as well on my G4 systems, usually I just use `dhclient` directly after it drops me to the root shell. This should give you connectivity on LAN. If you need Wifi, there is currently no straight up way without manually touching wpa_supplicant's configuration.

**Reply by iamhumanipromise** [reply]
Thank you! This solved the ethernet stuff much more quickly xD. 
Time then syncs! Whoo! 

---- 

Another noob issue: while following wiki to pacstrap, i recieve an error that _/mnt is out of space_. 

I then `mount -o remount; size=1G /run/archiso/cowspace` 

Attempt pacstrap again, but recieve the error that _/ is out of space_ 

PowerBook G4 has 1.25GB RAM so not sure how to best allocate space... or maybe instead, use mac-fdisk to create a temporary 2GB directory on the local HD and mount that as /mnt/? (Just a random thought) 

What would you recommend / suggest if you have seen this before? This is nothing POWER-specific I'm sure but I've not been able to Google-fu this specific situation on an x86 device to figure out how to apply it here. 

-----

**EDIT: SELF-RESOLVED by: 

- Cowspace size being 1256M not 1G
- ADDING partition swap partition type Linux_swap at block 24640 from the main instructions. Size = 3G otherwise runs out of space.  
- Then used mkswap /dev/sda3 followed by swapon /dev/sda3.** 

Could then proceed with pacstrap

**Reply by kth5** [reply]
Interesting, I have not encountered this on a powerbook G4 with 1GB RAM. Though I usually only install using:

`pacstrap /mnt base base-devel linux grub vim openssh networkmanager`

Depending on how many extra packages/groups you chose you may run out of space.

One workaround can be to either:
* arch-chroot into your mount to install more packages
* install base first and add packages from the booted system once you have space available in `/var/cache/pacman`
* make `/var/cache/pacman` a symlink to `<yourmount>/var/cache/pacman` when running the ISO
